### ISSUE TITLE
Modify Sample Code Based on Variables

### ISSUE THREAD

**Issue by wspeirs** [enzonotario/vitepress-openapi#92]
### What would you like?

The current version generates "static" code for each language, based only on the url and method. Unfortunately, the url is simply: `const url = props.baseUrl + props.path`. This has the issue that any path parameters are left as-is in the code.

It would be great if the sample code could be updated to include the variables in the "try it" box. This way the code sample would be "complete". I _think_ this would be done in much the same was as the `OACodeBlock` currently computes the URL for curl

### Why is this needed?

If you copy-and-paste the code sample as-is, you will end up with an invalid URL if there are path parameters.

Also, it would be nice to mix-and-match such that you can _remove_ the try it portion of the playground, but keep the sample code. This way people can "build" a request in their language of choice based on the variables form.

### How could it be implemented?

I _think_ you would need to extract (or just duplicate?) the code used to compute the curl information:

```
const curl = computed(() => {
  const headers = request.value.headers
  if (!headers?.['Content-Type']) {
    headers['Content-Type'] = 'application/json'
  }

  return fetchToCurl({
    method: props.method.toUpperCase(),
    url: request.value.url,
    headers,
    body: request.value.body ? formatJson(request.value.body) : null,
  })
})
```

Then pass this as a parameter to `generateCodeSamples` so that it can generate each code sample appropriately.

### Other information

It would also be _awesome_ to select _which_ languages you want to support, and possibly even some way to provide the template for each language. I realize this is above-and-beyond, but something to think about if you're changing this.

**Comment by enzonotario** [comment]
Hi @wspeirs I'm working on it in #116 , can you please validate it via [pkg-pr-new release](https://github.com/enzonotario/vitepress-openapi/runs/32565167087)?

It nows generates correct code for JavaScript/PHP/Python, and allows to customize lang list and the `generator` function, as mentioned in: https://vitepress-openapi-git-feature-cod-7dde87-enzo-notarios-projects.vercel.app/customizations/code-samples.html

Any feedback/suggestions are welcome!

**Comment by td-bill** [comment]
For a few (Bruno & cURL at least), it seems to be duplicating the param
![image](https://github.com/user-attachments/assets/f2ad0451-64fb-4dab-ac87-67486560e9b0)

**Comment by enzonotario** [comment]
Ups, I forgot to remove a duplicated process from other component. It's now centralized and working fine so far:

![image](https://github.com/user-attachments/assets/1efed7b2-e2e5-4e55-8cbb-360389fec752)

Can you please re-validate from [pkg-pr-new release](https://github.com/enzonotario/vitepress-openapi/pull/116#issuecomment-2453688510)? 

Thanks for this!

**Comment by enzonotario** [comment]
It's now released in [v0.0.3-alpha.48](https://github.com/enzonotario/vitepress-openapi/releases/tag/v0.0.3-alpha.48).

Thanks for reporting, any feedback is welcome!
